workflow-service/src/test/java/com/redhat/parodos/workflow/execution/service/WorkFlowServiceImplTest.java

```diff
@@ -394,6 +401,101 @@ void testUpdateWorkFlowTask() {
+		workFlowDefinition.setId(workFlowCheckerDefinitionId);
+		// workflow checker task definition
+		WorkFlowTaskDefinition workFlowCheckerTaskDefinition = WorkFlowTaskDefinition.builder()
+				.workFlowDefinition(workFlowDefinition).name(workFlowCheckerTaskName).build();
+		workFlowCheckerTaskDefinition.setId(UUID.randomUUID());
+		Mockito.when(this.workFlowTaskDefinitionRepository.findFirstByNameAndWorkFlowDefinitionType(Mockito.any(),
+				Mockito.any())).thenReturn(workFlowCheckerTaskDefinition);
+
+		// workflow checker task execution
+		WorkFlowExecution workFlowCheckerExecution = WorkFlowExecution.builder().status(WorkFlowStatus.IN_PROGRESS)
+				.projectId(projectId).workFlowDefinitionId(workFlowCheckerDefinitionId).build();
+		workFlowCheckerExecution.setId(UUID.randomUUID());
+		Mockito.when(this.workFlowRepository.findByMasterWorkFlowExecution(Mockito.any()))
+				.thenReturn(List.of(workFlowCheckerExecution));
+
+		WorkFlowTaskExecution workFlowTaskExecution = WorkFlowTaskExecution.builder().arguments("test").results("res")
+				.status(WorkFlowTaskStatus.FAILED).workFlowTaskDefinitionId(workFlowCheckerTaskDefinition.getId())
+				.workFlowExecutionId(workFlowCheckerExecution.getId()).build();
+		workFlowTaskExecution.setId(UUID.randomUUID());
+		Mockito.when(this.workFlowTaskRepository.findByWorkFlowExecutionIdAndWorkFlowTaskDefinitionId(Mockito.any(),
+				Mockito.any())).thenReturn(List.of(workFlowTaskExecution));
+
+		this.workFlowService.updateWorkFlowCheckerTaskStatus(workFlowExecutionId, workFlowCheckerTaskName,
+				WorkFlowTaskStatus.COMPLETED);
+
+		// then
+		ArgumentCaptor<WorkFlowTaskExecution> argument = ArgumentCaptor.forClass(WorkFlowTaskExecution.class);
+		Mockito.verify(this.workFlowTaskRepository, Mockito.times(1)).save(argument.capture());
+		assertEquals(argument.getValue().getWorkFlowTaskDefinitionId(), workFlowCheckerTaskDefinition.getId());
+		assertEquals(argument.getValue().getWorkFlowExecutionId(), workFlowCheckerExecution.getId());
+		assertEquals(argument.getValue().getStatus().toString(), "COMPLETED");
+	}
+
+	@Test
+	void testUpdateWorkFlowCheckerTaskStatusWithInvalidExecutionData() {
+		// given
+		UUID workFlowExecutionId = UUID.randomUUID();
+		String workFlowCheckerTaskName = "testWorkFlowTask";
+
+		// when
+		Mockito.when(this.workFlowRepository.findById(any())).thenReturn(Optional.empty());
+
+		// then
+		assertThrows(ResponseStatusException.class, () -> {
+			this.workFlowService.updateWorkFlowCheckerTaskStatus(workFlowExecutionId, workFlowCheckerTaskName,
+					WorkFlowTaskStatus.COMPLETED);
+		});
+
+		Mockito.verify(this.workFlowTaskRepository, Mockito.times(0)).save(any());
+	}
+
+	@Test
+	void testUpdateWorkFlowCheckerTaskStatusWithInvalidTaskData() {
+		// given
+		UUID workFlowExecutionId = UUID.randomUUID();
+		String workFlowCheckerTaskName = "testWorkFlowTask";
+
+		// when
+		Mockito.when(this.workFlowRepository.findById(Mockito.any()))
```

Comment:
If it's possible, I think it's better to use `Mockito.eq(....)` instead of `Mockito.any()` to make our unit tests more precise and to ensure that there aren't unwanted invocation. 
This suggestion applies also for other occurrences.
WDYT?